dagster_polars/io_managers/delta.py

```diff
@@ -180,20 +179,37 @@ def load_input(self, context: InputContext) -> Union[Any, Dict[str, Any]]:
                         " or Any: is '{type_annotation}'."
                     )
 
-    def dump_df_to_path(
+    def sink_df_to_path(
         self,
         context: OutputContext,
-        df: pl.DataFrame,
+        df: pl.LazyFrame,
         path: "UPath",
         metadata: Optional[StorageMetadata] = None,
     ):
-        assert context.metadata is not None
+        if context.metadata is None:
+            raise ValueError("You need to provide metadata to the asset.")
```

Comment:
this seems like a misleading error; `context.metadata` will always be populated for assets, the user can't do anything about it. 

A simple `assert context.metadata is not None` to satisfy `pyright` would be more clear here.

Reply:
Yeah at the time of writing this piece I wasn't sure whether a context.metadata is always there or not.

Reply:
I am removing the asserts an will use context_metadata = context.metadata or {}, got some other feedback from Robert to not use asserts